### ISSUE TITLE
`drush state:set system.maintenance_mode 1`: Too many arguments, expected arguments "command" "key".

### ISSUE THREAD

**Issue by ASauer** [drud/ddev#3230]
https://gist.github.com/ASauer/791d4bd955fab3980ff3bd7869259e56

I'm trying to run

`ddev drush state:set system.maintenance_mode 1`

With MAMP `drush state:set system.maintenance_mode 1` works fine. With ddev I get the error:

`Too many arguments, expected arguments "command" "key".`

I've tried every possible variation of this command I could think of, also using `ddev ssh`.
`ddev exec "drush state:set system.maintenance_mode 1"`
And all possible variations I could think of. Nothing works.

Am I missing something? Is this impossible with ddev?

- Host computer OS and Version:
OSX 10.15.7
- Docker Desktop version if on macOS or Windows (from "About Docker Desktop")
Version 4.0.0 (4.0.0.12)
- ddev version information (use `ddev version`)
ddev version v1.17.7
- config.yaml contents for the misbehaving project
name: sauermusic
type: drupal9
docroot: web
php_version: "7.4"
webserver_type: apache-fpm
router_http_port: "80"
router_https_port: "443"
xdebug_enabled: false
additional_hostnames: []
additional_fqdns: []
mariadb_version: ""
mysql_version: "5.7"
use_dns_when_possible: true
composer_version: ""
web_environment: []
- Do you have any custom configuration (nginx, php, mysql) in the .ddev folder? If so, have you tried without them?
No

**Comment by rfay** [comment]
The real question is what drush version you have. `ddev drush --version`.

What happens when you use your command `drush state:set system.maintenance_mode 1` inside the. container?

`ddev ssh` and then `drush state:set system.maintenance_mode 1` ?

**Comment by ASauer** [comment]
$ ddev drush --version
Drush Commandline Tool 10.6.0

$ ddev ssh drush state:set system.maintenance_mode 1
Error: accepts at most 1 arg(s), received 4
Usage:
  ddev ssh [projectname] [flags]
etc.

**Comment by ASauer** [comment]
or first
$ ddev ssh
then
$ drush state:set system.maintenance_mode 1                                       
  Too many arguments, expected arguments "command" "key".

**Comment by rfay** [comment]
OK, so this has nothing to do with ddev. Looks like you're using an invalid drush command, or else you've encountered a drush bug. But most likely you have the wrong command.

**Comment by rfay** [comment]
I imagine you have drush 8 in your path on your host computer on MAMP, and perhaps this command works with drush8. On MAMP, do a `drush --version` to see.

**Comment by rfay** [comment]
Dup of https://github.com/drush-ops/drush/issues/4818 and apparently https://github.com/drush-ops/drush/issues/4816

**Comment by ASauer** [comment]
I have only been using drush for a few weeks... I did a ddev powerdown. Started MAMP and ran drush --version:
Drush Commandline Tool 10.5.0.
This is the first version of drush that was ever installed on this computer.

I've tested it on the Drupal site that was made before I installed drush or ddev.
This site uses MAMP. (Drush is globally installed.)

It works without ddev, so the issue is with ddev, I guess.

**Comment by rfay** [comment]
You already showed that it does not work without ddev. Again, your answer seems to be in drush-land, please see https://github.com/drud/ddev/issues/3230#issuecomment-915622546

You already proved in https://github.com/drud/ddev/issues/3230#issuecomment-915603384 that it has nothing to do with ddev.

**Comment by ASauer** [comment]
I think I showed the opposite [15 minutes ago]((https://github.com/drud/ddev/issues/3230#issuecomment-915661693))
It works with MAMP after a powerdown of ddev.
No ddev, works, ddev, does not work.

I've just looked at the other comments and @greg-1-anderson does not think it is a ddev problem.
Another person had problems with `consolidation/annotated-command`.
It's not in my `composer.json` file.
Someone found a workaround:
`drush ev '\Drupal::state()->set("system.maintenance_mode", 1)'`
So drush in ssh mode works (I've tested it).

Or there is something I completely misunderstood... 
So please explain why, 
- if it is working while using MAMP 
- and it is working with the above mentioned workaround
, it is still a drush-problem???

**Comment by rfay** [comment]
The point is... if you `ddev ssh` and use drush... that is not using ddev at all. You're in a modern VM/container using the drush that is built into your project. Probably in MAMP, as I said, the project drush was not in your PATH and you were using drush8. 

You showed in https://github.com/drud/ddev/issues/3230#issuecomment-915603384 that drush inside the container with the same command failed. (`ddev ssh` and then give the command)

I imagine if you look at `drush --version` in MAMP you'll find it's a drush8 version, not the one that is built into your project.

**Comment by greg-1-anderson** [comment]
This is a bug in annotated-command 4.3.0, fixed in version 4.3.1.  Ensure that your Drush dependencies are all up-to-date and your problem should go away. Using Drush 10 you should always have a site-local Drush; if you are using a global Drush, install a site-local Drush and it will be used in preference to the global one.

**Comment by rfay** [comment]
I imagine you can `ddev composer update drush/drush --with-all-dependencies`

You could also just remove drush and re-add it, `ddev drush remove drush/drush` and then `ddev drush require drush/drush`

**Comment by ASauer** [comment]
@rfay Thanks for the clarification!
@greg-1-anderson For a newbie it is sometimes hard to make a link between dependencies you think you do not have, and finding the right command to check the issue, but thanks for updating annotated-command!

Problem solved:
I've removed any trace of a global drush and composer files and folders.
Also removed anything pointing to composer by checking .bash_profile, .bashrc and .profile (I'm on a mac).

Then I ran `ddev composer update drush/drush --with-all-dependencies`
The `--with-all-dependencies` is the important bit!
During the install I saw that `annotated-command` was updated to 4.3.1!
I did not know I had it installed... So, if you have a problem and someone mentions dependencies, try this!

`ddev restart`
essential!

Now it works!
Thanks for your patience and help :)
